nifi-docker/dockerhub/DockerBuild.sh

```diff
@@ -13,24 +15,16 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-#!/bin/bash
-
-DOCKER_UID=1000
-if [ -n "$1" ]; then
-  DOCKER_UID="$1"
-fi
+set -e
+set -o pipefail
 
-DOCKER_GID=1000
-if [ -n "$2" ]; then
-  DOCKER_GID="$2"
-fi
+DOCKER_UID="${1:-1000}"
+DOCKER_GID="${2:-1000}"
+MIRROR="${3:-https://archive.apache.org/dist}"
+BASE="${4:-${MIRROR}}"
```

Comment:
Can I suggest we do the mirror handling for the user (either here or in the Dockerfile itself)? For example, you can determine whether to go to archive.a.o or not by checking the in_dist from the Apache closer.lua script:

```sh
curl -sSLo - "https://www.apache.org/dyn/closer.lua/nifi/${NIFI_VERSION}/nifi-${NIFI_VERSION}-bin.zip?as_json&cca2=us" | \
    jq -r '(if .in_dist then .preferred else "https://archive.apache.org/dist/" end) + .path_info'
```

Reply:
That adds a jq dependency on the builder's machine (unless we do that in the Dockerfile, although that means installing the dependencies before setting some of the ARG/ENV, and that feels wrong)

Also, in_dist appears to always be false (chewing 1.13.2, 1.14.p, 1.15.0 and 1.15.0-SNAPSHOT), so we'll always build from the default archive site, which won't contain the in-progress RC artifacts. Unless I'm missing something (I've got no idea what the link is that you've referenced, so don't know what creates/updates it, etc.)